spec/features/user_registration_spec.rb

```diff
@@ -69,40 +70,40 @@ def signup(username, email, password)
  
 describe "Signing in" do
 
-describe "successfully" do
-  let!(:user) { FactoryGirl.create(:user) } 
-  it "displays a flash message indicating success" do
-    visit user_session_path #visit users/sign_in page
-    fill_in 'Email',    with: user.email
-    fill_in 'Password', with: user.password
-    click_button "Sign In"
-    expect(current_path).to eq "/"
-    expect(page).to have_content("Hello")
+  describe "successfully" do
+    let!(:user) { FactoryGirl.create(:user) } 
+    it "displays a flash message indicating success" do
+      visit user_session_path #visit users/sign_in page
```

Comment:
So, I was supposed to get you this:

https://github.com/plataformatec/devise/wiki/How-To:-Test-with-Capybara

This is how you log in without having to go through all the motions.

you will probably want to include the Warden::Test::Helpers module in the rails_helper file that's in your spec directory. But only for feature specs. Let me know if you need help figuring out how to do that. I can point you in the right direction.